### PR TITLE
Erc20 in da place!

### PR DESCRIPTION
**Pull request by vrolland**

### CODE REVIEW

File: packages/requestNetworkSmartContracts/contracts/synchrone/RequestCurrencyContractInterface.sol [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import '../core/RequestCore.sol';
+import '../base/math/SafeMath.sol';
+import '../base/math/SafeMathInt.sol';
+import '../base/math/SafeMathUint8.sol';
+
+
+/**
+ * @title RequestCurrencyContractInterface
+ *
+ * @dev RequestCurrencyContractInterface is the currency contract managing the request in Ethereum
```
Comment (romaric-juniet) [inline packages/requestNetworkSmartContracts/contracts/synchrone/RequestCurrencyContractInterface.sol]: Update description

File: packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+import '../core/RequestCore.sol';
+import '../base/math/SafeMathUint8.sol';
+import '../base/token/ERC20.sol';
+import './RequestCurrencyContractInterface.sol';
+import './RequestERC20Collect.sol';
+
+/**
+ * @title RequestERC20
+ *
+ * @dev RequestERC20 is the currency contract managing the request in Ethereum
```
Comment (romaric-juniet) [inline packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol]: Update

File: packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol [hunk 3]
```diff
@@ -0,0 +1,602 @@
...
+ *
+ * @dev Requests can be created by the Payee with createRequestAsPayee(), by the payer with createRequestAsPayer() or by the payer from a request signed offchain by the payee with broadcastSignedRequestAsPayer
+ */
+contract RequestERC20 is RequestCurrencyContractInterface, RequestERC20Collect {
+	using SafeMath for uint256;
+	using SafeMathInt for int256;
+	using SafeMathUint8 for uint8;
+
+	// payment addresses by requestId (optional). We separate the Identity of the payee/payer (in the core) and the wallet address in the currency contract
+    mapping(bytes32 => address[256]) public payeesPaymentAddress;
```
Comment (romaric-juniet) [inline packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol]: broken indentation in the file

File: packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+		// set payment address for payer
+		if(_payerRefundAddress != 0) {
+			payerRefundAddress[requestId] = _payerRefundAddress;
+		}
+
+		return requestId;
+	}
+
+	/*
+	 * @dev Function to broadcast and accept an offchain signed request (can be paid and additionals also)
```
Comment (romaric-juniet) [inline packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol]: Clarify the comments "(can be paid and additionals also)"

File: packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol [hunk 5]
```diff
@@ -0,0 +1,602 @@
...
+
+		// extract the number of payees
+		uint8 payeesCount = uint8(_requestData[40]);
+		int256 totalExpectedAmounts = 0;
+		for(uint8 i = 0; i < payeesCount; i++) {
+			// extract the expectedAmount for the payee[i]
+			int256 expectedAmountTemp = int256(extractBytes32(_requestData, uint256(i).mul(52).add(61)));
+			// compute the total expected amount of the request
+			totalExpectedAmounts = totalExpectedAmounts.add(expectedAmountTemp);
+			// all expected amount must be positibe
```
Comment (romaric-juniet) [inline packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol]: positive

File: packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol [hunk 6]
```diff
@@ -0,0 +1,602 @@
...
+	function paymentAction(
+		bytes32 _requestId,
+		uint256[] _payeeAmounts,
+		uint256[] _additionalAmounts)
+		external
+		whenNotPaused
+	{
+		// automatically accept request if request is created and msg.sender is payer
+		if (requestCore.getState(_requestId)==RequestCore.State.Created && msg.sender == requestCore.getPayer(_requestId)) {
+			acceptInternal(_requestId); // TODO more costy than just call requestCore
```
Comment (romaric-juniet) [inline packages/requestNetworkSmartContracts/contracts/synchrone/RequestERC20.sol]: Why is it a TODO?
typo: costly
remove commented out code

**Comment by vrolland** [comment]
close to push on development

**Pull request closed without merging**